core/server/scheduling/post-scheduling/index.js

```diff
@@ -0,0 +1,99 @@
+var Promise = require('bluebird'),
+    moment = require('moment'),
+    utils = require(__dirname + '/../utils'),
+    events = require(__dirname + '/../../events'),
+    errors = require(__dirname + '/../../errors'),
+    models = require(__dirname + '/../../models'),
+    schedules = require(__dirname + '/../../api/schedules'),
+    _private = {};
+
+_private.normalize = function normalize(options) {
+    var object = options.object,
+        apiUrl = options.apiUrl,
+        client = options.client;
+
+    return {
+        time: object.get('published_at'),
+        url: apiUrl + '/schedules/posts/' + object.get('id') + '?client_id=' + client.get('slug') + '&client_secret=' + client.get('secret'),
+        extra: {
+            httpMethod: 'PUT',
+            oldTime: object.updated('published_at') || null
+        }
+    };
+};
+
+_private.loadClient = function loadClient() {
+    return models.Client.findOne({slug: 'ghost-scheduler'}, {columns: ['slug', 'secret']});
+};
+
+_private.loadScheduledPosts = function () {
+    return schedules.getScheduledPosts({
+        from: moment().subtract(7, 'days').startOf('day').toDate(),
+        to: moment().endOf('day').toDate()
+    }).then(function (result) {
+        return result.posts || [];
+    });
+};
+
+exports.init = function init(options) {
+    options = options || {};
+
+    var config = options.postScheduling,
+        apiUrl = options.apiUrl,
+        adapter = null,
+        client = null;
+
+    if (!config) {
+        return Promise.reject(new errors.IncorrectUsage());
+    }
+
+    if (!apiUrl) {
+        return Promise.reject(new errors.IncorrectUsage());
+    }
+
+    return _private.loadClient()
+        .then(function (_client) {
+            client = _client;
+
+            return utils.createAdapter(config);
+        })
+        .then(function (_adapter) {
+            adapter = _adapter;
+
+            return _private.loadScheduledPosts();
```

Comment:
`loadScheduledPosts` is also triggered when we run it on `Ghost(Pro)`.
This has the advantage of integrity and has the disadvantage that we maybe just reinsert the jobs.
We cannot move this piece of logic into the `SchedulingDefault`, because this adapter can be used also for newsletter scheduling. So i just kept it here and we can discuss and decide.